app/src/main/java/campus/tech/kakao/map/SearchActivity.kt

```diff
@@ -79,37 +80,20 @@ class SearchActivity : AppCompatActivity() {
             binding.textView.visibility =
                 if (placesAdapter.itemCount <= 0) View.VISIBLE else View.GONE
         })
-
-        viewModel.searchHistoryData.observe(this, Observer {  searchHistoryData ->
-            searchHistoryList = searchHistoryData
+        
+        viewModel.searchHistoryData.observe(this, Observer { searchHistoryData ->
+            searchHistoryAdapter.submitList(searchHistoryData.toList())
+            binding.searchHistory.isVisible = searchHistoryData.isNotEmpty()
         })
     }
 
-    private fun updateSearchHistoryVisibility() {
-        binding.searchHistory.isVisible = searchHistoryList.isNotEmpty()
-    }
-
-    private fun searchHistoryContains(itemName: String): Int {
-        return searchHistoryList.indexOfFirst { it.word == itemName }
-    }
-
-    private fun moveSearchToLast(foundIdx: Int, itemName: String) {
-        viewModel.moveSearchToLast(foundIdx, itemName)
-        searchHistoryAdapter.notifyItemMoved(foundIdx, searchHistoryList.size - 1)
-    }
-
-    private fun insertSearch(search: String) {
-        val foundIdx = searchHistoryContains(search)
-        if (foundIdx != -1) {
-            moveSearchToLast(foundIdx, search)
-        } else {
-            viewModel.addSearch(search)
-            searchHistoryAdapter.notifyItemInserted(searchHistoryList.size)
+    private fun goToSearch(place: Place) {
+        val intent = Intent(this, MainActivity::class.java).apply {
+            putExtra("place", place)
         }
-    }
-
-    private fun delSearch(position: Int) {
-        viewModel.delSearch(position)
-        searchHistoryAdapter.notifyItemRemoved(position)
+        Log.d("searchAct State", "Intent is: $intent")
+        intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_SINGLE_TOP
+        startActivity(intent)
```

Comment:
이 부분은 [1주차 피드백](https://github.com/kakao-tech-campus-2nd-step2/android-contacts/pull/19)에도 남겨드렸던 내용입니다만, ActivityResult API를 통해 startAcitivity를 다시 호출하지 않고도, MainActivity로 되돌아 가면서 place가 담긴 Intent를 돌려줄 수 있게 작성해주시면 좋겠습니다.

```suggestion
        setResult(Activity.RESULT_OK, intent)
        // setResult에 담긴 intent를 MainActivity에서 다시 받으려면 
        // startActivity(Intent(this, SearchActivity::class.java)) 부분도 수정이 필요합니다.
```

ActivityResult는 데이터 교환이 필요한 액티비티 전환에 가장 기본적인 요소입니다.
지금처럼 바로 직전화면으로 돌아가는 경우에 다시 startActivity를 이용하는 것은 올바른 화면 전환 방법이 아닙니다.